### PR TITLE
Add logaddexp2 and xlogy pointwise operators

### PR DESCRIPTION
**Pull request by zheng1**
## Summary

Adds two missing ATen pointwise operators implemented in Triton:

- **`logaddexp2`** — `log2(2**x + 2**y)`, computed in the numerically stable
  form `m + log2(1 + 2**(-|x - y|))` where `m = max(x, y)`. Mirrors the existing
  `logaddexp` operator.
- **`xlogy`** — `x * log(y)`, following PyTorch's ATen semantics in this
  precedence: `NaN` if `y` is `NaN`; `0` where `x == 0`; otherwise `x * log(y)`.

Both are implemented with `@pointwise_dynamic` (broadcast / dtype-promotion /
`out=` handled automatically), registered in `flag_gems` (including the `.out`
variants), and added to the operator inventory `conf/operators.yaml` as `beta`.

## Motivation

`torch.logaddexp2` and `torch.xlogy` are standard pointwise ops (xlogy is common
in KL-divergence / entropy computations) that were not yet covered by FlagGems.
`logaddexp2` is the natural companion to the already-supported `logaddexp`.

## Correctness

Accuracy tests compare against the PyTorch reference across all
`POINTWISE_SHAPES` × `{fp16, fp32, bf16}`, plus a dedicated `xlogy` special-value
test covering `x == 0`, `y == 0`, and `y == NaN`.

```
tests/test_logaddexp2.py tests/test_xlogy.py .... 75 passed
```

Verified on **NVIDIA A100-PCIE-40GB** (torch 2.6.0+cu124, triton 3.2.0).

## Performance

Binary-pointwise benchmarks added for both ops. As memory-bound elementwise
kernels, they track `torch` closely (Gems speedup ~1.0–1.17 across shapes on
A100), e.g. `xlogy_out` fp32:

| shape | torch (ms) | gems (ms) | speedup |
|---|---|---|---|
| 64×64 | 0.00717 | 0.00614 | 1.17 |
| 4096×4096 | 0.1587 | 0.1577 | 1.01 |
| 1024³ | 9.416 | 9.315 | 1.01 |

## Notes

- For non-finite inputs, `logaddexp2(±inf, ±inf)` returns `NaN` (from
  `inf - inf`), matching the behavior of the existing `logaddexp` implementation.
  Happy to special-case `inf` if the maintainers prefer stricter parity.
- `xlogy` implements the `Tensor`/`OutTensor` overloads; scalar overloads fall
  through to the default backend.

## Checklist

- [x] Operators implemented in Triton via `pointwise_dynamic`
- [x] Registered (+ `.out` variants) and added to `conf/operators.yaml`
- [x] Accuracy tests (all shapes × fp16/fp32/bf16 + special values)
- [x] Benchmarks
- [x] `isort --profile black` / `black` / `flake8` clean

### CODE REVIEW

**Comment by tengqm** [comment]
@zheng1 Please sign the CLA in order for your PR to be reviewed.

File: src/flag_gems/ops/xlogy.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    xlogy_func_tensor_scalar(self, other, out0=out)
+    return out
+
+
+def xlogy_scalar_tensor(self, other):
+    logger.debug("GEMS XLOGY_SCALAR_TENSOR")
+    return xlogy_func_scalar_tensor(self, other)
+
+
+def xlogy_scalar_tensor_out(self, other, out):
```
Comment (huangyiqun) [inline src/flag_gems/ops/xlogy.py:68]: Please add all operator interfaces in this file, including xlogy, xlogy_out, xlogy_tensor_scalar, xlogy_tensor_scalar_out, xlogy_scalar_tensor and xlogy_scalar_tensor_out, to conf/operators.yaml. Also, please add accuracy and performance tests with the corresponding @pytest.mark.{op_name} decorators.

File: src/flag_gems/__init__.py [hunk 2]
```diff
@@ -447,6 +447,14 @@ def torch_ge(v):
...
     ("logaddexp", logaddexp),
     ("logaddexp.out", logaddexp_out),
+    ("logaddexp2", logaddexp2),
+    ("logaddexp2.out", logaddexp2_out),
+    ("xlogy.Tensor", xlogy),
+    ("xlogy.OutTensor", xlogy_out),
+    ("xlogy.Scalar_Other", xlogy_tensor_scalar),
+    ("xlogy.Scalar_Self", xlogy_scalar_tensor),
+    ("xlogy.OutScalar_Other", xlogy_tensor_scalar_out),
+    ("xlogy.OutScalar_Self", xlogy_scalar_tensor_out),
```
Comment (huangyiqun) [inline src/flag_gems/__init__.py]: Please sort alphabetically.

**Comment by zheng1** [comment]
Thanks for the review! I have addressed all three points in the latest commit:

- Registered all six `xlogy` interfaces (`xlogy`, `xlogy_out`, `xlogy_tensor_scalar`, `xlogy_tensor_scalar_out`, `xlogy_scalar_tensor`, `xlogy_scalar_tensor_out`) in `conf/operators.yaml`.
- Gave each interface its own `@pytest.mark.{op_name}`, added accuracy tests for the out-scalar overloads, and added performance tests for all four scalar interfaces.
- Sorted the `xlogy`/`logaddexp2` entries alphabetically in `_FULL_CONFIG`.

All accuracy tests pass locally (`--ref=cpu --quick` and full run) and `pre-commit run --all-files` is clean. Please take another look -- thanks!

**Review by bin913 — APPROVED** [review]
